docs/src/imagesegmentation.md

```diff
@@ -0,0 +1,317 @@
+# ImageSegmentation.jl
+
+## Introduction
+
+Image Segmentation is the process of partitioning the image into regions that have similar attributes. Image segmentation has various applications e.g, medical image segmentation, image compression and is used as a preprocessing step in higher level vision tasks like object detection and optical flow. This package is a collection of image segmentation algorithms written in Julia.
+
+## Installation
+
+```julia
+Pkg.add("ImageSegmentation")
+```
+
+## Example
+
+Let's see an example on how to use the segmentation algorithms in this package. We will try to seperate the deer, the ground and the sky in the image below using seeded region growing algorithm. This algorithm initializes the regions as the user-provided seed points and iteratively adds pixels to each region.
+
+![Original](assets/segmentation/deer.jpg) 
+
+```julia
+using Images, ImageSegmentation, ImageView
+
+img = load("deer.jpg")
```

Comment:
@waldyrious The problem with linking to Wikimedia Commons page is the size of the image. I require a smaller version of that image so that it looks good on the tutorials page. For the `seeded_region_growing`, I need to mention some hard- coded co-ordinates of the image as seeds so that the user can try it if he wants. Is it possible to use this image from imgbb.com and add a reference to the original source of the image (Wikimedia) as a footnote?

Reply:
I'll reply in the main PR thread for convenience, since this discussion is happening on an outdated code block that github collapses by default.